examples/stm32/f0/other/can/can.c

```diff
@@ -0,0 +1,182 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/cm3/systick.h>
+#include <libopencm3/stm32/can.h>
+
+
+#define TS1FROMBTRVAL(value)	(value & 0xF0000) //((1<<16)|(1<<17)(1<<18)(1<<19)))
+#define TS2FROMBTRVAL(value)	(value & 0x700000)//((1<<20)|(1<<21)(1<<22)))
+#define SJWFROMBTRVAL(value)	(value & 0x3000000)//((1<<24)|(1<<24)))
+#define BRPFROMBTRVAL(value)	((value & 0x1FF)+1)
+
+ /* 500kbit/s -> CAN_BTR = 0x001c0005 http://www.bittiming.can-wiki.info/ */
+#define BTR500k 0x001c0005
+
+struct can_tx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+};
+
+struct can_rx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+	uint8_t fmi;
+};
+
+struct can_tx_msg can_tx_msg;
+struct can_rx_msg can_rx_msg;
+
+
+static void systick_setup(void)
+{
+	/* 72MHz / 8 => 9000000 counts per second */
+	systick_set_clocksource(STK_CSR_CLKSOURCE_AHB);
+
+	/* 9000000/9000 = 1000 overflows per second - every 1ms one interrupt */
+	/* SysTick interrupt every N clock pulses: set reload to N-1 */
+	systick_set_reload(8999);
+
+	systick_interrupt_enable();
+
+	/* Start counting. */
+	systick_counter_enable();
+}
+
+static void can_setup(void)
+{
+	/* Enable peripheral clocks. */
+
+	//B8 Rx
+	//B9 Tx
+	rcc_periph_clock_enable(RCC_GPIOB);
+
+	rcc_periph_clock_enable(RCC_CAN);
+
+	/* Configure CAN pin: RX Alternate Function (input pull-up). */
+	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_PULLUP, GPIO8);
+	gpio_set_af(GPIOB, GPIO_AF4, GPIO8);
+
+	/* Configure CAN pin: TX Alternate Function */
+	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_NONE, GPIO9);
+	gpio_set_af(GPIOB, GPIO_AF4, GPIO9);
+
+	/* Reset CAN. */
+	can_reset(CAN1);
+
+	/* CAN cell init. */
+	if (can_init(CAN1,
+		     false,           /* TTCM: Time triggered comm mode? */
+		     true,            /* ABOM: Automatic bus-off management? */
+		     false,           /* AWUM: Automatic wakeup mode? */
+		     true,           /* NART: No automatic retransmission? */
+		     false,           /* RFLM: Receive FIFO locked mode? */
+		     false,           /* TXFP: Transmit FIFO priority? */
+		     SJWFROMBTRVAL(BTR500k),
+			 TS1FROMBTRVAL(BTR500k),
+		     TS2FROMBTRVAL(BTR500k),
+			 BRPFROMBTRVAL(BTR500k),
+		     false,				 /* Loopback */
+		     false))             /* Silent */
+	{
+		/* can init returned a non zero value
+		 * this can be caused by wrong configuration of the alternate function pins,
+		 * or other reasons keeping the can module from reading 11 recessive bits for
+		 * leaving initialization phase
+		 */
+		__asm__("bkpt");
```

Comment:
does the same. while(1) {} behaves strange with my board debugging using SWD (might be the stlink)
changed to while(1) {}